superset/views/base.py

```diff
@@ -170,16 +170,61 @@ def json_response(self, obj, status=200):
             mimetype="application/json",
         )
 
+    def menu_data(self):
```

Comment:
This may not be a real concern (and probably outside the scope of this PR), but it seems like we're doing a fair amount of work here at each request. On the FAB side we should make sure that this method doesn't lead to more than a single database round trip, and I'm wondering whether we'd want to cache any of this, probably not.